app/code/Magento/PageCache/Plugin/Framework/View/Element/InlineFormKeyUpdater.php

```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ *
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+ 
+namespace Magento\PageCache\Plugin\Framework\View\Element;
+
+
+class InlineFormKeyUpdater
+{
+    public function afterToHtml(\Magento\Framework\View\Element\FormKey $subject, $result)
+    {
+        return $result.$this->getInlineJavaScript();
+    }
+    
+    private function getInlineJavaScript()
+    {
+        $uniqueId = uniqid();
+        
+        return '<script id="'.$uniqueId.'">document.getElementById("'.$uniqueId.'").previousSibling.value = formKey;</script>';
```

Comment:
As Magento introduced CSP, we need to replace this code with something like this:
https://github.com/magento/magento2/blob/89d3cda99975c0db2967165311184be84d26e9b4/app/code/Magento/Theme/view/frontend/templates/page/js/require_js.phtml#L15